lib/Sema/CSStep.h

```diff
@@ -630,7 +666,7 @@ class TypeVariableStep final : public BindingStep<TypeVarBindingProducer> {
     // at a current batch of bindings is done it's a
     // success because each new batch would be less
     // and less precise.
-    return AnySolved && needsToComputeNext();
+    return solvedAnyWithoutFixes() && needsToComputeNext();
```

Comment:
This is interesting, which test-cases are improved by this change?

Reply:
This improves the diagnostic for

https://github.com/apple/swift/pull/30686/files#diff-b0ad7ac038888d7fabf47adde33127adR283-R286

as we now explore the unwrapped type of the argument as a binding.

Reply:
Thanks! I asked because I was wondering whether it makes more sense to re-compute certain types while inferring bindings instead of relying on this check. I feel like in diagnostic mode we'd want to always try bindings exhaustively...

Reply:
@xedin I'm not sure I follow – do you mean moving the logic from `TypeVarBindingProducer::computeNext` to `determineBestBindings`?

Reply:
Yeah, either moving logic to `determineBestBindings` or ignoring `needsToComputeNext` in diagnostic mode, which is probably safer.

Reply:
Ah okay, the updated code does the latter :) How does it look?

Reply:
Looks good, thank you!

Reply:
Thanks for reviewing!

Reply:
Thanks for your help with diagnostics!